File: reactapp/src/services/folderService.js
import apiClient from './apiClient';

export const getFoldersByOwner = (ownerId) => apiClient.get(`/api/folders/user/${ownerId}`);

export const getChildFolders = (folderId) => apiClient.get(`/api/folders/${folderId}/children`);

export const createFolder = (ownerId, name, parentId = null) =>
  apiClient.post('/api/folders', null, { params: { ownerId, name, parentId } });

export const renameFolder = (folderId, ownerId, name) =>
  apiClient.patch(`/api/folders/${folderId}/rename`, { name }, { params: { ownerId } });

export const deleteFolder = (folderId, ownerId) =>
  apiClient.delete(`/api/folders/${folderId}`, { params: { ownerId } });

File: springapp/src/main/java/com/examly/springapp/service/FolderService.java
package com.examly.springapp.service;

import com.examly.springapp.model.Folder;
import java.util.List;

public interface FolderService {
    Folder createFolder(String name, Long ownerId, Long parentId);
    Folder renameFolder(Long folderId, String newName, Long ownerId);
    void deleteFolder(Long folderId, Long ownerId);
    List<Folder> getRootFolders(Long ownerId);
    List<Folder> getAllFolders(Long ownerId);
    /** Alias used by FolderController */
    List<Folder> getFoldersByOwner(Long ownerId);
    /** Alias used by FolderController */
    List<Folder> getChildFolders(Long parentId);
}
